### PR TITLE
issue/1934: updated modernizr

### PR DESCRIPTION
**Pull request by oliverfoster**
#1934 

Works done:
* Have retrofitted ``Modernizr.touchevent`` to ``Modernizr.touch``
* Have added ``.touch`` and ``.no-touch`` for backward compatibility 
* Replaced ``Modernizr.load`` with ``loadScript`` function 
* Created ``Adapt.loadScript`` API
* Removed Modernizr checks for <IE11
* Switched core to use new ``.no-touchevents`` and ``.touchevents``  classes instead of using ``.touch`` and ``.no-touch``
* Switch core to use Adapt.device.touch rather than Modernizr.touch

Breaking changes:
* Dropped ``Modernizr.load`` API use ``Adapt.loadScript`` instead

### CODE REVIEW

File: src/core/js/scriptLoader.js [hunk 1]
```diff
@@ -34,24 +45,27 @@
 
     // 3. start loading JQuery, wait for it to be loaded
     function loadJQuery() {
-        Modernizr.load([
-            {
-                load: 'libraries/jquery.v2.min.js',
-                complete: checkJQueryStatus
-            }
-        ]);
+        loadScript('libraries/jquery.v2.min.js', checkJQueryStatus);
```
Comment (brian-learningpool) [inline src/core/js/scriptLoader.js]: Isn't this altered by the switch to JQuery v3?

Comment (oliverfoster) [reply src/core/js/scriptLoader.js]: Yup, but they are two separate PRs, so I'll hit that merge conflict when I get there,

File: src/core/js/scriptLoader.js [hunk 2]
```diff
@@ -34,24 +45,27 @@
...
-            loadFoundationLibraries();
+            setupModernizr();
         }
     }
-    
-    //5. Load foundation libraries and templates then load Adapt itself
+
+    //5. Backward compatibility for Modernizr
+    function setupModernizr() {
+        Modernizr.touch = Modernizr.touchevents;
```
Comment (brian-learningpool) [inline src/core/js/scriptLoader.js]: Is the declaration of the `touch` property for backward compatibility between v2 and v3?  

In this PR you also refer to `Adapt.device.touch`.  Should this be the preferred property for plugins to check after the bootstrapping?

Comment (oliverfoster) [reply src/core/js/scriptLoader.js]: Yes, it's for backward compatibility only. I don't know if there are any other plugins which reference ```Mozernizr.touch```

Comment (brian-learningpool) [reply src/core/js/scriptLoader.js]: Thanks.  I can't think of a core plugin but I guess there could be non-core ones using it.

File: src/core/js/scriptLoader.js [hunk 3]
```diff
@@ -1,5 +1,16 @@
 (function() {
 
+    function loadScript(url, callback){
+        if (!url || !(typeof url === 'string')) return;
```
Comment (tomgreenfield) [inline src/core/js/scriptLoader.js]: Am I missing the reason behind using brackets instead of the below?
```js
typeof url !== 'string'
```

**Pull request merged**